internal/envoy/secrets.go

```diff
@@ -301,29 +301,38 @@ func (s *secretManager) unwatch(names []string, node string) error {
 }
 
 // Manage is used for re-fetching expiring TLS certificates and updating them
```

Comment:
Some extra context about what this function is doing in the docstring here would be helpful I think. It's not immediately obvious that we're checking on an interval to see if our cached cert has expired and also force-refreshing our cached cert on a separate interval.